pkg/api/getambassador.io/v2/handwritten.conversion.go

```diff
@@ -658,3 +660,52 @@ func Convert_v3alpha1_TCPMappingSpec_To_v2_TCPMappingSpec(in *v3alpha1.TCPMappin
 
 	return nil
 }
+
+func Convert_v2_TracingServiceSpec_To_v3alpha1_TracingServiceSpec(in *TracingServiceSpec, out *v3alpha1.TracingServiceSpec, s conversion.Scope) error {
+	if err := autoConvert_v2_TracingServiceSpec_To_v3alpha1_TracingServiceSpec(in, out, s); err != nil {
+		return err
+	}
+	// WARNING: in.TagHeaders requires manual conversion: does not exist in peer-type
+	// if only tag_headers are set, translate to custom_tags.
+	// if both are set, ignore tag_headers.
```

Comment:
In theory I agree but I think the issue is that this logic executes on the `apiext` container which means it would get logged there and not in the emissary container where users would expect to see this type of warning. For that reason, I'm not so sure it is necessary and debugging using snapshots would yield a similar result.

However, this information will be important to capture in the docs so I will be sure to add a some thing on this.

Reply:
ok but I find it a little inconsistent that `Convert_v3alpha1_TracingServiceSpec_To_v2_TracingServiceSpec` logs a warning when both `TagHeaders` and `CustomHeaders` are set and `TagHeaders` are ignored but we don't log when converting from v2 to v3alpha1.

Presumably both are logged in the `apiext` container?

Reply:
I think logging a warning would be nice so that any users why might have apiext log aggregation happening could potentially pick up on the error without having to run through the snapshot (or just for any emissary users who don't have support to help them review a snapshot), but that said, nothing else in the conversion up until now logs anything so since it is not an established pattern I'm not going to ask you to hold this PR for that change.

Reply:
I think that is fair, since I have to rebase on master anyways to fix the merge conflicts I will add a log line.

Reply:
I have rebased to address merge conflicts and add a new commit 44464d31a74eec5de232f5bb551af3d36c3db5bd that logs the warning so that we are consistent.